backend/src/apis/round.rs

```diff
@@ -0,0 +1,226 @@
+use ethers::{
+    abi::Address,
+    types::{Bytes, U256},
+};
+use halo2_proofs::{
+    halo2curves::bn256::{Bn256, Fr as Fp, G1Affine},
+    plonk::{ProvingKey, VerifyingKey},
+    poly::kzg::commitment::ParamsKZG,
+};
+use serde_json::to_string_pretty;
+use snark_verifier_sdk::CircuitExt;
+use std::error::Error;
+
+use crate::contracts::{generated::summa_contract::summa::Asset, signer::SummaSigner};
+use summa_solvency::{
+    circuits::{
+        merkle_sum_tree::MstInclusionCircuit,
+        solvency::SolvencyCircuit,
+        utils::{full_prover, gen_proof_solidity_calldata, generate_setup_artifacts},
+    },
+    merkle_sum_tree::MerkleSumTree,
+};
+
+pub(crate) type SetupArtifacts = (
+    ParamsKZG<Bn256>,
+    ProvingKey<G1Affine>,
+    VerifyingKey<G1Affine>,
+);
+
+#[derive(Debug, Clone)]
+pub struct SolvencyProof {
+    public_inputs: Vec<U256>,
+    proof_calldata: Bytes,
+}
+
+impl SolvencyProof {
+    pub fn get_public_inputs(&self) -> &Vec<U256> {
+        &self.public_inputs
+    }
+
+    pub fn get_proof_calldata(&self) -> &Bytes {
+        &self.proof_calldata
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct MstInclusionProof {
+    public_inputs: Vec<Vec<Fp>>,
+    proof: Vec<u8>,
+}
+
+impl MstInclusionProof {
+    pub fn get_public_inputs(&self) -> &Vec<Vec<Fp>> {
+        &self.public_inputs
+    }
+
+    pub fn get_proof(&self) -> &Vec<u8> {
+        &self.proof
+    }
+}
+
+pub struct Snapshot<const LEVELS: usize, const N_ASSETS: usize, const N_BYTES: usize> {
```

Comment:
Add `assets_state` to the Snapshot, it will containt `[Asset; N_ASSETS]`

Reply:
Updated at [3d1ab95](https://github.com/summa-dev/summa-solvency/pull/152/commits/3d1ab95db73fcfe38006aa7f300b3a78f15fd678)